src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1652,6 +1660,11 @@ ExpressionPtr Visitor::visit([[maybe_unused]] Parser::BuiltInCallContext* ctx) {
     return createUnary(&makeRoundExpression);
   } else if (functionName == "floor") {
     return createUnary(&makeFloorExpression);
+  } else if (functionName == "if") {
+    return createTernary(&makeIfExpression);
+  } else if (functionName == "coalesce") {
+    AD_CORRECTNESS_CHECK(ctx->expressionList());
+    return makeCoalesceExpression(visit(ctx->expressionList()));
```

Comment:
Why is this not a more generic `createVariadic(&makeCoalesceExpression)`? This connects to the earlier question about why the `CoalesceExpression` implementation is so (relatively) complex.

Reply:
It could be, maybe I'll do it like this. But there is only one other variadic expression in this function (CONCAT), because (`[NOT] IN`) is part of a different grammar rule.